crates/re_viewport/src/context_menu/actions.rs

```diff
@@ -375,14 +377,147 @@ impl ContextMenuAction for MoveContentsToNewContainerAction {
+                    for (identifier, display_name) in space_view_classes
+                        .iter()
+                        .map(|identifier| {
+                            (
+                                identifier,
+                                space_view_class_registry
+                                    .get_class_or_log_error(identifier)
+                                    .display_name(),
+                            )
+                        })
+                        .sorted_by_key(|(_, display_name)| display_name.to_owned())
+                    {
+                        if ui.button(display_name).clicked() {
+                            create_space_view_with_entities(ctx, *identifier);
+                        }
+                    }
+                };
+
+            ui.label(egui::WidgetText::from("Recommended:").italics());
+            if recommended_space_view_classes.is_empty() {
+                ui.label("None");
+            } else {
+                buttons_for_space_view_classes(ui, &recommended_space_view_classes);
+            }
+
+            ui.label(egui::WidgetText::from("Others:").italics());
+            buttons_for_space_view_classes(ui, &other_space_view_classes);
+        })
+        .response
+    }
+}
+
+/// Builds a list of compatible space views for the provided selection.
+fn recommended_space_views_for_selection(
+    ctx: &ContextMenuContext<'_>,
+) -> IntSet<SpaceViewClassIdentifier> {
+    re_tracing::profile_function!();
+
+    let entities_of_interest = ctx
+        .selection
+        .iter()
+        .filter_map(|(item, _)| item.entity_path())
+        .collect::<Vec<_>>();
+
+    let mut output: IntSet<SpaceViewClassIdentifier> = IntSet::default();
+
+    let space_view_class_registry = ctx.viewer_context.space_view_class_registry;
+    let entity_db = ctx.viewer_context.entity_db;
+    let applicable_entities_per_visualizer =
+        space_view_class_registry.applicable_entities_for_visualizer_systems(entity_db.store_id());
+
+    for entry in space_view_class_registry.iter_registry() {
+        let visualizable_entities = determine_visualizable_entities(
+            &applicable_entities_per_visualizer,
+            entity_db,
+            &space_view_class_registry.new_visualizer_collection(entry.class.identifier()),
+            &*entry.class,
+            &EntityPath::root(),
+        );
+
+        // We consider a space view class to be recommended all selected entities are "visualizable"
+        // with it. By "visualizable" we mean that either the entity itself, or any of its
+        // sub-entities, are visualizable.
+
+        let covered = entities_of_interest.iter().all(|entity| {
+            visualizable_entities.0.iter().any(|(_, entities)| {
+                entities
+                    .0
+                    .iter()
+                    .any(|e| e == *entity || e.is_descendant_of(entity))
+            })
+        });
```

Comment:
[not actionable]
oof that's a lot of entities to walk through if the user happens to select many. Ah well it happens only on-click and what are the odds of the user selecting everything.. right.. ;-)
We need to get establish better patterns & algorithms using prefix tree operations for things like this :thinking:

Reply:
Hopefully `all`'s short-circuit should help us in most cases 🤞🏻